### ISSUE TITLE
Access optional output of HoughLines depending on input datatype

### ISSUE THREAD

**Issue by ddrobny** [skvark/opencv-python#164]
### Expected behaviour

The function HoughLines() allows to determine the returned value by passing either a 2D or 3D vector as the output container. From the function documentation:
```
    .   @param lines Output vector of lines. Each line is represented by a 2 or 3 element vector
    .   \f$(\rho, \theta)\f$ or \f$(\rho, \theta, \textrm{votes})\f$ . \f$\rho\f$ is the distance from the coordinate origin \f$(0,0)\f$ (top-left corner of
    .   the image). \f$\theta\f$ is the line rotation angle in radians (
    .   \f$0 \sim \textrm{vertical line}, \pi/2 \sim \textrm{horizontal line}\f$ ).
    .   \f$\textrm{votes}\f$ is the value of accumulator.
```

According to the C++ code I should be able to pass a 3D vector to trigger the returned object to have 3 elements for each detected line (shape (n, 1, 3) where n is the number of detected lines).

### Actual behaviour

I tried passing numpy ndarrays of different dimension but this had no effect on the return value. I always end up with a ndarray of shape (n, 1, 2) where n is the number of detected lines.

### Steps to reproduce
```
returned_lines = np.ndarray((1, 3))    # also tried returned_lines = np.ndarray((1, 1, 3))  and other
lines = cv2.HoughLines(img, 1, np.pi / 180,  0, lines=returned_lines)
```

- opencv-python version is 3.4.5.20

**Comment by ddrobny** [comment]
This behavior might be a problem for other functions as well, but this is what I found problematic for me. Can the wrapper parse any python datatype into the expected C++ type?

**Comment by skvark** [comment]
Please report the issue to the OpenCV repository (https://github.com/opencv/opencv/issues) or if this is a usage question ask it at http://answers.opencv.org/questions/.

This repository is just a collection of build scripts which generate pre-built packages automatically from the OpenCV sources. If there are bugs in OpenCV all fixes must be applied to the C++ code at the upstream.

**Comment by ddrobny** [comment]
This is no bug in the C++ code.

This is a problem of the python wrapper apparently not capturing the full functionality of the underlying C++ function.

Do you have any suggestions on how to improve the wrapper functionality?

**Comment by skvark** [comment]
I'm sorry, I don't have any. I suggest that you read https://docs.opencv.org/master/da/d49/tutorial_py_bindings_basics.html. Feature requests should be submitted also to the OpenCV repository.

**Comment by ddrobny** [comment]
All right, thank you!

**Comment by cher-nov** [comment]
Just to note. This was addressed in https://github.com/opencv/opencv/issues/17050 and seems to be fixed in 4.5.2 by https://github.com/opencv/opencv/pull/19423.

Usage example: https://github.com/opencv/opencv/blob/4.5.2/modules/python/test/test_houghlines.py#L67

Unfortunately, 4.5.2 is not on pypi at the moment - the latest version there is still 4.5.1.48.
